RetailCoder.VBE/Navigation/CodeExplorer/CodeExplorerViewModel.cs

```diff
@@ -26,29 +27,31 @@ public sealed class CodeExplorerViewModel : ViewModelBase, IDisposable
     {
         private readonly FolderHelper _folderHelper;
         private readonly RubberduckParserState _state;
+        private bool _sourceControlEnabled;
 
-        public CodeExplorerViewModel(FolderHelper folderHelper, RubberduckParserState state, List<CommandBase> commands)
+        public CodeExplorerViewModel(FolderHelper folderHelper, RubberduckParserState state, List<CommandBase> commands, IGeneralConfigService configLoader)
         {
             _folderHelper = folderHelper;
             _state = state;
             _state.StateChanged += HandleStateChanged;
             _state.ModuleStateChanged += ParserState_ModuleStateChanged;
+            _sourceControlEnabled = configLoader.LoadConfiguration().UserSettings.GeneralSettings.SourceControlEnabled;
```

Comment:
IIRC we're injecting `UserSettings`. not sure it's good to inject the `configLoader` instead

Reply:
Given this is set in the constructor, injecting the `UserSettings` could work (is the binding even setup for that?), but injecting the loader makes it possible to [eventuallty] honor modified settings without reloading RD entirely.

Reply:
Since we're talking about this...

I'd like to add the ability to save the view choices as a persisting setting... what's the best way to go about that?

Reply:
@retailcoder so... why don't we just make the Injection Points of `UserSettings` **listen** for changes and just bind the UserSettings `InSingletonScope`?  that would also solve the problem, wouldn't it?

@shadowofsilicon IIUC you can just add a property to the relevant Settings class i.e. the `WindowSettings` and "just wire it up" in the Settings Dialog. Persistence should be taken care of by Rubberduck.SettingsProvider.

Reply:
Take a close look re: Settings.  Completely overhauled it.